numpy/f2py/_backends/_meson.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import annotations
+
+import errno
+import shutil
+import subprocess
+from pathlib import Path
+
+from ._backend import Backend
+from string import Template
+
+import warnings
+
+
+class MesonTemplate:
+    """Template meson build file generation class."""
+
+    def __init__(
+        self,
+        modulename: str,
+        sources: list[Path],
+        deps: list[str],
+        object_files: list[Path],
+        linker_args: list[str],
+        c_args: list[str],
+    ):
+        self.modulename = modulename
+        self.build_template_path = (
+            Path(__file__).parent.absolute() / "src" / "meson.build.src"
```

Comment:
Can you choose another file extension name? `.src` has a specific meaning in numpy (see all the `.c.src` files) and this has nothing to do with that. `.template` or some such thing will do here.